### PR TITLE
longest substring without repetition of chars

### PR DESCRIPTION
**Pull request by pradeep98**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

[Python] longest substring without repetition of characters. Issue opened by @piyushgoell Assigned by @SKAUL05 

### Checklist

- [X] I've read the contribution guidelines.
- [X] I've checked the issue list before deciding what to submit.
- [X] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

[Longest Substring Without Repeating Characters #563](https://github.com/TesseractCoding/NeoAlgo/issues/563)

### CODE REVIEW

File: Python/other/longest_Substring.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+def longestSubstringLength(string):
```
Comment (SKAUL05) [inline Python/other/longest_Substring.py]: Document our function using docstrings

File: Python/other/longest_Substring.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        else:
+            if di[val] >= start:
+                start = di[val] + 1
+                counter = index - di[val] -1
+            counter += 1
+            di[val] = index
+        result = max(result, counter)
+    return result
+
+testCase = int(input('No. of test cases:'))
```
Comment (SKAUL05) [inline Python/other/longest_Substring.py]: Follow PEP8 Guidelines for variable naming & function naming conventions
https://www.python.org/dev/peps/pep-0008/#id45

**Review by SKAUL05 — CHANGES_REQUESTED** [review]

**Review by SKAUL05 — APPROVED** [review]
Looks Good

**Review by Kajol-Kumari — COMMENTED** [review]
```
No. of test cases:1                                                                                                       
Teststring                                                                                                                     
6      
```                                                                                                               
Hey @pradeep98 If i am getting it correctly then your code should have output `3` as at the 4th position we have a common character `s` but it showing `6`. can you please explain, why?

**Comment by pradeep98** [comment]
> ```
> No. of test cases:1                                                                                                       
> Teststring                                                                                                                     
> 6      
> ```
> 
> Hey @pradeep98 If i am getting it correctly then your code should have output `3` as at the 4th position we have a common character `s` but it showing `6`. can you please explain, why?
Ma'am, statement says: longest substring without repetition of characters. Here in Teststring->  substring 'string' is the longest one with no repetition of characters. Thats why answer is 6.

**Review by Kajol-Kumari — APPROVED** [review]
It's good to go :)

**Review by siddharth25pandey — APPROVED** [review]
@pradeep98 Looks fine to me.
Able to merge.
Thanks!

**Comment by iamrajiv** [comment]
@all-contributors please add @pradeep98 for code

**Pull request merged**